src/usePopper.js

```diff
@@ -0,0 +1,99 @@
+// @flow
+import { useLayoutEffect, useRef, useState, useMemo } from 'react';
+import {
+  createPopper as defaultCreatePopper,
+  type Options as PopperOptions,
+} from '@popperjs/core';
+
+type Options = { ...PopperOptions, createPopper: typeof defaultCreatePopper };
+
+type State = {
+  styles: {
+    [key: string]: $Shape<CSSStyleDeclaration>,
+  },
+  attributes: {
+    [key: string]: { [key: string]: string | boolean },
+  },
+};
+
+export const usePopper = (
+  referenceElement: Element,
+  popperElement: HTMLElement,
+  options: Options = {}
+) => {
+  const popperInstanceRef = useRef();
+  const [state, setState] = useState<State>({
+    styles: {
+      popper: { position: 'absolute', left: '0', top: '0' },
+    },
+    attributes: {},
+  });
+  const createPopper = useMemo(
+    () => options.createPopper || defaultCreatePopper,
+    [options.createPopper]
+  );
+  const popperOptions = useMemo(() => {
+    // eslint-disable-next-line no-unused-vars
+    const { createPopper, ...popperOptions } = options;
+    return popperOptions;
+  }, [options]);
+
+  const updateStateModifier = useMemo(
+    () => ({
+      name: 'updateState',
+      enabled: true,
+      phase: 'write',
+      fn: ({ state }) => {
+        const elements = Object.keys(state.elements);
+
+        setState({
+          styles: Object.fromEntries(
```

Comment:
I don't think we should use `fromEntries` due to old browsers, keep polyfills as minimal as `@popperjs/core`

Reply:
I doubt it's going to be an issue, those React libraries are usually consumed through webpack+babel